react/src/lib/components/ColorScales/ColorScales.tsx

```diff
@@ -15,14 +14,45 @@ const DEFAULT_SCALE = [
     "#2a4858",
 ];
 
+const propTypes = {
```

Comment:
Why PropTypes definition is needed here given this is a typescript react component?

Reply:
This is because Dash is not supporting TypeScript yet. The workflow is: 
1) TypeScript files get transpiled and the result output to `/dist`.
2) Dash used `react-docgen` to parse the output main component JavaScript files in `/dist`  in order to extract both, variables, their types and their description (doc string) from the `propTypes` object.
3) This extracted information is used to generate the respective Python files (variables, doc strings and types).
4) TypeScript is limited to check types at buildtime. When the application is running, TS types don't help us determine if a given variable value has a valid type. Here, `propTypes` come to the rescue, making sure that even at runtime, the values are always valid. Keep in mind that after building the application, we are only using a minified, webpack-packed JavaScript file.

Reply:
Thank you, I understand now. So the top level .tsx component would need to have propTypes defined to make it work with Dash, but is it necessary for nested components too? I saw for example, the Suggestions.tsx also has propTypes.

Reply:
No, you're right, I think it's not necessary to have them there since the parent component should take care of validating the types. It's a bit legacy from before TypeScript so we could probably remove them as long as we're not planning to use the components directly at some point (which is a quite unlikely scenario). For now, I would leave it as is and move this rather into a new issue. Agreed?